contributor-docs/operations/tests/rfc2119-keyword-test-guide.md

```diff
@@ -0,0 +1,173 @@
+
+## 2. Interpreting RFC2119 Keywords in the Context of SCS
+
+- **MUST and MUST NOT**: These are binary and straightforward to test. Compliance or
+  non-compliance directly affects whether the standard is satisfied. For visualization, fulfilled
+  requirements could be marked in green, unfulfilled ones in red.
+- **SHOULD and SHOULD NOT**: These represent recommendations. While fulfilling these do
+  not directly affect standard compliance, it is advised for future-proofing against
+  potential standard evolutions. Non-compliance could be visualized in yellow, possibly
+  with a valid reason for non-fulfillment.
+- **MAY**: This is more flexible and could be used to provide additional
+  information or best practices. Their impact on compliance is not as direct as **MUST** or
+  **SHOULD** categories. These can be marked as blue.
+
+## 3. Channels for Output in Test Scripts
+
+In test scripts, different channels are used to convey information of different importance to the user. These channels are based on the ubiquitous and de-facto standard logging levels of common logging libraries in scripting and programming languages, in particular Python: DEBUG, INFO, WARNING, ERROR and CRITICAL.
+
+Alignement of the RFC2119 keywords with specific channels in test scripts:
+
+- **MUST** and **MUST NOT** lead to the ERROR channel for failed tests
+- **SHOULD** and **SHOULD NOT** lead to the WARNING channel, the test is still passed
+- **MAY** leads to the INFO channel, the test is still passed
+
+However, the CRITICAL level is not directly associated with any of the RFC2119 keywords; it is instead used to signify that a test was unable to complete or was interrupted due to various issues, such as runtime failures.
+
+## 4. Compliance and Test Passing Criteria
+
+A test is considered to pass if and only if it doesn't produce any messages on the
+ERROR nor the CRITICAL channel. The presence of warnings or informational items from
+**SHOULD** and **MAY** categories do not directly impact the pass/fail status but is
+important for overall quality and compatibility with future standards.
+
+That means a test can have one of three results:
+
+- fail: if it has at least one message on ERROR
+- DNF (did not finish): if it has no message on ERROR, but at least one on CRITICAL
+- pass: otherwise
+
+In addition, a test **MUST** exit with a non-zero exit code (e.g., via `sys.exit(…)`) if there are any ERROR or CRITICAL messages, thus signaling a failure to meet a standard.
+
+It is crucial to determine and explicitly state within this document the expected conventions for logging output, specifically regarding the redirection of all channel outputs to standard error (**stderr**), which aligns with Python's default logging behavior. The format which **MUST** be used is **CHANNEL(MESSAGE)**, where **CHANNEL** represents the log level (e.g., DEBUG, INFO, WARNING, ERROR, CRITICAL) and **MESSAGE** encapsulates the actual log message.
```

Comment:
> It is crucial to determine and explicitly state within this document the expected conventions for logging output

This sounds like a prompt for the reader to determine the conventions and should be shortened to something like "the scripts are expected to ...".

Instead of `CHANNEL(MESSAGE)` the actual format we use is `CHANNEL: MESSAGE`. Example:

    INFO: Checking cluster specified by context 'my-cluster' in /path/to/.kube/config.

Reply:
done